fs/operations/operations.go

```diff
@@ -1519,7 +1600,23 @@ func moveOrCopyFile(ctx context.Context, fdst fs.Fs, fsrc fs.Fs, dstFileName str
 		return err
 	}
 
-	if NeedTransfer(ctx, dstObj, srcObj) {
+	NoNeedTransfer := false
+	if fs.Config.CompareDest != "" {
+		CompareDestDir, err := GetCompareDest()
+		if err != nil {
+			return err
+		}
+		NoNeedTransfer = CompareDest(ctx, dstObj, srcObj, CompareDestDir)
+	}
```

Comment:
This should probably be `} else if {` since `CompareDest` and `CopyDest` are mutually exclusive